src/Toml/Type/AnyValue.hs

```diff
@@ -31,7 +32,7 @@ instance Show AnyValue where
 instance Eq AnyValue where
     (AnyValue (Bool b1))    == (AnyValue (Bool b2))    = b1 == b2
     (AnyValue (Integer i1)) == (AnyValue (Integer i2)) = i1 == i2
-    (AnyValue (Double f1))  == (AnyValue (Double f2))  = f1 == f2
+    (AnyValue (Double f1))  == (AnyValue (Double f2))  = isRight $ sameValue (Double f1) (Double f2)
```

Comment:
The solution here is not completely correct... What I would like to see is that if you have `Anyvalue val1` and `AnyValue val2` you need to call `sameValue` on `val1` and `val2`, pattern match on the result and use `Eq` instance for `Value`. `sameValue` function only checks that types of values are the same, not the values themselves.